cap_client/cap_api.py

```diff
@@ -68,17 +68,21 @@ def _make_request(self,
                               stream=stream,
                               **kwargs)
 
-        if response.status_code == expected_status_code:
+        if stream:
+            resp_data = response
+        else:
             try:
-                resp_data = response if stream else response.json()
+                resp_data = response.json()
             except ValueError:
-                resp_data = None
-            return resp_data
-        else:
+                resp_data = response.text if response.text else None
+
+        if response.status_code != expected_status_code:
             raise BadStatusCode(endpoint=endpoint,
                                 expected_status_code=expected_status_code,
                                 status_code=response.status_code,
-                                data=response.json())
+                                data=resp_data)
```

Comment:
sorry, but this doesn't work :( 

it happens, because in the case of text response, your code will pass a string in `data` param for `BadStatusCode`, and it will fail on trying to construct an error message inside a class constructor, this line [here](https://github.com/cernanalysispreservation/cap-client/blob/69ccd308147aaa097621725e83b8cb9b579e4419/cap_client/errors.py#L46).

Thats what you would see if you tried to run your code:
<img width="1041" alt="Screenshot 2020-03-20 at 09 09 19" src="https://user-images.githubusercontent.com/21820788/77146281-f663fb00-6a8a-11ea-852d-f4dd0700ad06.png">